### PR TITLE
Write lesson on debugging the user interaction program

### PR DESCRIPTION
**Pull request by kcleland0818**

### CODE REVIEW

**Review by linda-seiter — COMMENTED** [review]
Before debugging a program that involves user input, can we step through debugging a program that has a logic error?  The way this lesson is written, it seems like the debugger is what you would use when dealing with exceptions.  In fact, maybe add a lesson covering debugging after the operators lesson, then have the second half of this lesson here showing how to debug for exceptions.

File: README.md [hunk 1]
```diff
@@ -1,32 +1,322 @@
...
+
+We performed integer division. As a reminder, with integer division, the decimal
+part is discarded or removed so that we only have an `integer` result. This is
+performed when we have two `int` values being divided.
+
+So `pizzaSlices / people` is the same as saying `10 / 3`, which evaluates to 3
+when using integer division. If we were to perform real division, `10 / 3` would
+be approximately 3.33.
+
+Let's say we had a pizza cutter where we could cut slice of pizza into thirds.
```
Comment (developerjaym) [inline README.md]: Consider adding 'a' between 'cut' and 'slice'

File: src/main/java/BuggyProgram.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+public class BuggyProgram {
+    public static void main (String[] args) {
+
+        int pizzaSlices = 10;
+        int people;
+
+        // How many slices of pizza should each person get?
+        int slices = pizzaSlices / people;
```
Comment (developerjaym) [inline src/main/java/BuggyProgram.java]: Consider renaming 'slices' to 'slicesPerPerson'

**Review by developerjaym — APPROVED** [review]

**Review by linda-seiter — APPROVED** [review]
Nice lesson!

**Comment by kcleland0818** [comment]
I changed the variable `slices` to `slicesPerPerson` recommended by Jay. This involved changing some words and screenshots. Double check I caught them all please?

File: README.md [hunk 3]
```diff
@@ -1,32 +1,322 @@
...
-## Topic 2
+    int pizzaSlices = 10;
+    int people;
 
-Discuss the topic of the reading. Break into sections and sub-sections as
-appropriate to help with the reader's orientation with the material and future
-referencing.
+    // How many slices of pizza should each person get?
+    int slicesPerPerson = pizzaSlices / people;
+    System.out.println("Each person should have " + slices + " slices of pizza.");
```
Comment (developerjaym) [inline README.md]: Should this be slicesPerPerson?

(Sorry!)

**Review by developerjaym — CHANGES_REQUESTED** [review]
Looks like one 'slices' didn't get updated

**Review by developerjaym — APPROVED** [review]

**Review by linda-seiter — APPROVED** [review]

**Pull request merged**